core/src/main/java/hudson/util/ColorPalette.java

```diff
@@ -40,6 +40,8 @@
     public static final Color YELLOW = new Color(0xFC,0xE9,0x4F);
     public static final Color BLUE = new Color(0x72,0x9F,0xCF);
     public static final Color GREY = new Color(0xAB,0xAB,0xAB);
+    public static final Color DARK_GREY = new Color(0x77,0x77,0x77);
+    public static final Color LIGHT_GREY = new Color(0xcc,0xcc,0xcc);
```

Comment:
Oh, so it's this file that causes weird GUIs to appear in macOS.

Reply:
@jvz WDYM?

Reply:
There seems to be a bug on macOS where referencing any AWT classes in a headless application will spawn a minimal GUI environment which is pretty much just an icon in the dock along with an application menu at the top. It steals focus from you while running the tests because macOS has yet to discover the joy of not allowing focus stealing. >:(